lib/blockscore/base.rb

```diff
@@ -0,0 +1,99 @@
+require 'blockscore/connection'
+
+module BlockScore
+  class Base
+    extend Connection
+
+    attr_reader :attributes
+
+    def initialize(options = {})
+      @attributes = options
+    end
+
+    def inspect
+      "#<#{self.class}:0x#{object_id.to_s(16)} JSON: " + JSON.pretty_generate(attributes)
+    end
+
+    def refresh
+      r = self.class.retrieve(id)
+      @attributes = r.attributes
+
+      true
+    rescue Error
+      false
+    end
+
+    def save
+      save!
+    rescue
+      false
+    end
+
+    def save!
+      self.class.post self.class.endpoint, attributes
+
+      true
+    end
+
+    def self.resource
+      @resource ||= Util.to_underscore(to_s.split('::').last)
+    end
+
+    def self.api_url
+      'https://api.blockscore.com/'
+    end
+
+    def self.endpoint
+      if self == Base
```

Comment:
Style/IfUnlessModifier: Favor modifier `if` usage when having a single-line body. Another good alternative is the usage of control flow `&&`/`||`.